agent/lib/kontena/service_pods/creator.rb

```diff
@@ -38,9 +38,9 @@ def perform
           end
         end
         service_config = service_pod.service_config
-        if service_pod.overlay_network
-          Celluloid::Actor[:network_adapter].modify_create_opts(service_config)
-        end
+
+        Celluloid::Actor[:network_adapter].modify_create_opts(service_config)
```

Comment:
The only thing it configures is the `HostConfig` `Dns`, `DnsSearch`, `DnsOptions` using the `docker0` interface IP.

Those should work for both host network and any kontena service containers... although it might strictly be better to use the `docker_gwbridge` IP?